src/NLog/Layouts/JsonLayout.cs

```diff
@@ -73,6 +75,18 @@ public JsonLayout()
         /// </summary>
         public bool RenderEmptyObject { get; set; }
 
+
+        /// <summary>
+        /// Gets or sets the option to include all properties from the log events
+        /// </summary>
+        public bool IncludeAllProperties { get; set; }
+
+        /// <summary>
+        /// List of property names to exclude when IncludeAllProperties is true
+        /// </summary>
+        public IList<string> ExcludedProperties { get; set; }
```

Comment:
I don't this will work from the XML, yet! 

See https://github.com/NLog/NLog/pull/1439 (but I won't support IList). 

Is it possible to write an unit test for this, so we won't forget it? (and it will succeed if we rebase after https://github.com/NLog/NLog/pull/1439 has been finished)